crates/hotshot/src/lib.rs

```diff
@@ -218,12 +221,19 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> SystemContext<TYPES, I> {
         );
 
         let mut saved_leaves = HashMap::new();
-        let mut saved_block_payloads = BlockPayloadStore::default();
+        let mut saved_payload_commitments = PayloadCommitmentStore::default();
```

Comment:
This naming seems off to me. We're not saving payload commitments. We're saving payloads, _indexed by_ their commitments. I think just `PayloadStore`/`saved_payloads` would be better

Reply:
Oh, we are actually saving the encoded transaction bytes instead of the payload, since it's a commitment -> bytes mapping. I'll rename the variable and struct.

Reply:
On second thought, `PayloadStore` and `saved_payloads`are indeed better. We are mapping the payload commitment to its encoded format, so it's basically a store for payloads.